app/controllers/profile_controller.rb

```diff
@@ -9,6 +13,35 @@ def show
     @user = current_user
   end
 
+  def password_form
+    @password_form = ChangePasswordForm.new({})
+  end
+
+  def change_password
+    @password_form = ChangePasswordForm.new(params[:change_password_form])
+    if @password_form.valid?
+      backend_change_password(
+        old: @password_form.old_password,
+        proposed: @password_form.password,
+        access_token: current_user.access_token
+      )
+      flash[:notice] = t('profile.password_changed')
+      redirect_to profile_path
+    else
+      flash.now[:errors] = @password_form.errors.full_messages.join(', ')
+      render :password_form, status: :bad_request
+    end
+  rescue InvalidOldPasswordError
+    flash[:error] = t('profile.old_password_mismatch')
+    render :password_form, status: :bad_request
+  rescue InvalidNewPasswordError
+    flash[:error] = t('devise.sessions.InvalidPasswordException')
+    render :password_form, status: :bad_request
+  rescue AuthenticationBackendException
+    flash[:error] = t('devise.failure.unknown_cognito_response')
+    render :password_form, status: :bad_request
+  end
```

Comment:
I think using a `return` and an `ensure` can get rid of all the `render'`s

```suggestion
  def change_password
      .....
      redirect_to profile_path
      return
    else
      flash.now[:errors] = @password_form.errors.full_messages.join(', ')
    end
  rescue InvalidOldPasswordError
    flash[:error] = t('profile.old_password_mismatch')
  rescue InvalidNewPasswordError
    flash[:error] = t('devise.sessions.InvalidPasswordException')
  rescue AuthenticationBackendException
    flash[:error] = t('devise.failure.unknown_cognito_response')
  ensure
    render :password_form, status: :bad_request
  end
```

Reply:
Did this and it didn't work... The ensure overrides the return.